### DISCUSSION TITLE
How to enable variable-pitch for consult-org-heading used with vertico?

### DISCUSSION THREAD

**Discussion by IceAsteroid** [minad/consult#1102] [Q&A] [answered]
Hi, good days, everyone.

Is it possible to enable `variable-pitch` specifically for `consult-org-heading` that's used with the `vertico` completion but not affecting other completions of `vertico`?

**Comment by hmelman** [comment]
FWIW `consult-outline` for me uses the faces from the buffer (variable pitch for me in markdown) while `consult-imenu` uses the regular minibuffer face.  I don't think I'm doing anything in my config to enable this. I don't use `consult-org-heading`, but maybe that can point you in a useful direction.

**Reply by IceAsteroid** [reply]
But it seems not working in my environment, at least for `consult-outline`, I gotta do more investigation.

**Reply by hmelman** [reply, marked as answer]
So to be clear, in `markdown-mode` I do enable `variable-pitch-mode` but I also have it configured so that headers use markdown-mode faces like `markdown-header-face-1`, I assume org can do similarly (e.g., `org-level-2`).  For me, `consult-outline` shows the headers in `markdown-header-face-1` or whichever one is on the face text-property of the header in the markdown buffer.

**Reply by IceAsteroid** [reply]
Thank you for the suggestion, I got it working!

It seems that `org-level-*` faces need to be explicitly specified with attributes, so `consult-org-heading` can parse the attributes properly in the minibuffer with `vertico`.

However, the font size shrinks a bit in the minibuffer, that's fine. I don't wanna further mess up the `consult-org.el` code or interfere the size that's already fine for the headings in the `org-mode` buffer, if there's not a better way here. 

----

One attempt I've tried before is adding `variable-pitch-mode` in the hook `vertico-mode-hook`, but this doesn't work for some reason.

Here's the code [Fix typo]:
```
;;; Mixed-pitch configuration
(defun my-setup/mixed-pitch ()
  ;;(with-eval-after-load 'buffer-face
  (setq default-frame-alist nil)
  (custom-theme-set-faces
   'user
   '(default ((t (:font "JetBrainsMono Nerd Font-12"))))
   '(variable-pitch ((t (:font "Dosis-v2031b" :height 1.2))))
   '(fixed-pitch ((t (:font "JetBrainsMono Nerd Font" :height 0.8))))
   '(line-number ((t (:inherit fixed-pitch :height 1.0 :slant normal))))
   '(line-number-current-line ((t (:inherit fixed-pitch :height 1.0 :slant normal)))))

  (defun my-setup/org-mixed-pitch ()
    (with-eval-after-load 'org
      (custom-set-faces
       '(org-document-title ((t (:font "Linux Libertine Display G" :height 2.0 :underline t))))
       '(org-block ((t (:inherit fixed-pitch))))
       '(org-code ((t (:inherit (shadow fixed-pitch)))))
       '(org-document-info-keyword ((t (:inherit (shadow fixed-pitch)))))
       '(org-indent ((t (:inherit (org-hide fixed-pitch)))))
       '(org-meta-line ((t (:inherit (font-lock-comment-face fixed-pitch)))))
       '(org-property-value ((t (:inherit fixed-pitch))) t)
       '(org-special-keyword ((t (:inherit (font-lock-comment-face fixed-pitch)))))
       '(org-table ((t (:inherit fixed-pitch))))
       '(org-tag ((t (:inherit (shadow fixed-pitch) :weight bold :height 0.8))))
       '(org-verbatim ((t (:inherit (shadow fixed-pitch))))))

      (dotimes (count 9)
	(when (< 0 count)
	  (set-face-attribute
	   (intern (concat "org-level-" (number-to-string count))) nil
	   :inherit (intern (concat "outline-" (number-to-string count)))
	   :font (face-attribute 'variable-pitch :family)
	   :height 'unspecified
	   :weight 'bold)))
      (add-hook 'org-mode-hook 'variable-pitch-mode))
    )
  (my-setup/org-mixed-pitch)
  )
(my-setup/mixed-pitch)

```

Here's the demo:
Before:
![consult-org-heading-variable-pitch-discussion2](https://github.com/user-attachments/assets/d3a3770d-8aa6-41a6-820f-739d4ae49c58)

After:
![consult-org-heading-variable-pitch-discussion1](https://github.com/user-attachments/assets/603612ef-99c0-43dc-b3bd-fd459e43188b)

**Reply by IceAsteroid** [reply]
Much more readable now, that's good!
